src/Core/Content/Product/ProductException.php

```diff
@@ -102,4 +103,20 @@ public static function reviewNotActive(): self|ReviewNotActiveExeption
             'Reviews not activated'
         );
     }
+
+    /**
+     * @deprecated tag:v6.8.0 - reason:return-type-change - Will only return `self` in the future
+     */
+    public static function originalIdNotFound(string $originalId): self|\RuntimeException
+    {
+        if (!Feature::isActive('v6.8.0.0')) {
+            return new \RuntimeException(\sprintf('Cannot find originalId %s in listing mapping', $originalId));
```

Comment:
for RuntimeExceptions deprecation is not necessary. the DomainExceptions deep down extend from RuntimeException, so who ever caught that, will still be able to catch this new exception.